nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -159,6 +166,95 @@ impl CrucibleDisk {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct LocalStorageDisk {
+    pub disk: model::Disk,
+    pub disk_type_local_storage: DiskTypeLocalStorage,
+    pub local_storage_dataset_allocation: Option<LocalStorageDatasetAllocation>,
+}
+
+impl LocalStorageDisk {
+    /// Create a new unallocated LocalStorageDisk
+    pub fn new(
+        disk: model::Disk,
+        disk_type_local_storage: DiskTypeLocalStorage,
+    ) -> LocalStorageDisk {
+        LocalStorageDisk {
+            disk,
+            disk_type_local_storage,
+            local_storage_dataset_allocation: None,
+        }
+    }
+
+    pub fn model(&self) -> &model::Disk {
+        &self.disk
+    }
+
+    pub fn id(&self) -> Uuid {
+        self.model().id()
+    }
+
+    pub fn name(&self) -> &api::external::Name {
+        self.model().name()
+    }
+
+    pub fn time_deleted(&self) -> Option<DateTime<Utc>> {
+        self.model().time_deleted()
+    }
+
+    pub fn project_id(&self) -> Uuid {
+        self.model().project_id
+    }
+
+    pub fn runtime(&self) -> DiskRuntimeState {
+        self.model().runtime()
+    }
+
+    pub fn state(&self) -> DiskState {
+        self.model().state()
+    }
+
+    pub fn size(&self) -> model::ByteCount {
+        self.model().size
+    }
+
+    pub fn slot(&self) -> Option<u8> {
+        self.model().slot()
+    }
+
+    pub fn required_dataset_overhead(&self) -> external::ByteCount {
+        self.disk_type_local_storage.required_dataset_overhead()
+    }
+
+    /// Return the full path to the local storage zvol's device
+    pub fn zvol_path(&self) -> Result<String, Error> {
```

Comment:
nit, unimportant: why is this a `String` rather than a `Utf8Path`?

Reply:
we're going to introduce more restrictions for dataset and volume names in a future PR, going to revisit this there